NebulaHost/PacketProcessors/Factory/Entity/CreatePrebuildsRequestProcessor.cs

```diff
@@ -44,6 +44,10 @@ public void ProcessPacket(CreatePrebuildsRequest packet, NebulaConnection conn)
                     planet.physics = new PlanetPhysics(planet);
                     planet.physics.Init();
                 }
+                if (AccessTools.Field(typeof(CargoTraffic), "beltRenderingBatch").GetValue(planet.factory.cargoTraffic) == null)
```

Comment:
https://harmony.pardeike.net/api/HarmonyLib.AccessTools.html#HarmonyLib_AccessTools_FieldRefAccess__2___0_System_Reflection_FieldInfo_

To avoid allocations.

Reply:
So instead:
```cs
AccessTools.Field(typeof(CargoTraffic), "beltRenderingBatch").GetValue(planet.factory.cargoTraffic)
```

having:
```cs
AccessTools.FieldRef(typeof(CargoTraffic), "beltRenderingBatch").GetValue(planet.factory.cargoTraffic)
```

right?

Reply:
FieldRef is typed. You need to cache the result of FieldRef to benefit from it though.

Reply:
Here's an example: https://github.com/hubastard/nebula/blob/183eee5f00096db8358aef26f408bba42fd4b695/NebulaWorld/Factory/FactoryManager.cs#L65-L69